app/src/org/commcare/android/database/user/models/FormRecordV3.java

```diff
@@ -0,0 +1,93 @@
+package org.commcare.android.database.user.models;
```

Comment:
This class needs to replace all references to the previous "FormRecord" class in the UserDatabaseUpgrader that existed, in addition to being used in the current upgrader.

`upgradeSixteenSeventeen() `, for example, should actually now be upgrading from `FormRecordV2` to `FormRecordV3`

Reply:
Ugh yea sorry, fixed

Reply:
By the way, as far as I can tell, `upgradeSixteenSeventeen()` is the only place this needed to happen. The only other reference to `FormRecord` is in `markSenseIncompleteUnsent()`, but that is being used by really old migrations, so I assume something must be different about that, or it would have had to change a long time ago